### PR TITLE
Fix mutation cache updates | Closes #2584

### LINKED ISSUE #2584: Default price(s) are added to Apollo cache multiple times upon ticket creation
Steps to reproduce:
- Create a default price if you don't have one yet.
- Make sure that the default price is related to an existing ticket.
- Reload EDTR to make sure both the ticket and its default price is loaded.
- Create a new ticket
- Check the Apollo Cache to see the default price duplicated or Open TPC for the newly created Ticket to see the price repeated.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Fixes duplication of default prices on Ticket creation
- Fixes/improves mutation cache updates
- Improves Apollo queries to make the keys consistent
- Closes #2584

### CODE REVIEW

File: assets/src/domain/eventEditor/services/apollo/initialization/useCacheRehydration.ts [hunk 1]
```diff
@@ -2,28 +2,27 @@ import { useEffect } from 'react';
 import { pathOr } from 'ramda';
 
 import useCacheRehydrationData from './useCacheRehydrationData';
-import useRelations from '../../../../../application/services/apollo/relations/useRelations';
-import { useStatus, TypeName } from '../../../../../application/services/apollo/status';
+import { useRelations } from '@appServices/apollo/relations';
+import { useStatus, TypeName } from '@appServices/apollo/status';
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/initialization/useCacheRehydration.ts]: soo satisfying :smiley:

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/updateTicketCache.ts [hunk 2]
```diff
@@ -27,7 +24,16 @@ const updateTicketCache = ({ proxy, datetimeIn, datetimeId, remove = false }: Ca
 		return;
 	}
 
-	const newDatetimeIn = remove ? datetimeIn.filter((id: string) => id !== datetimeId) : [...datetimeIn, datetimeId];
+	let newDatetimeIn: typeof datetimeIn;
+
+	switch (action) {
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/updateTicketCache.ts]: where is `default` ? :)

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/services/apollo/mutations/datetimes/updateTicketCache.ts]: No default needed

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useOnDeleteDatetime.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ const useOnDeleteDatetime = (): DatetimeMutationCallbackFn => {
 			});
 		}
 		// Update datetime cache after tickets cache is updated.
-		updateDatetimeCache({ proxy, datetimes, datetime, remove: true });
+		updateDatetimeCache({ proxy, datetimes, datetime, action: 'remove' });
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useOnDeleteDatetime.ts]: good one, `action: 'remove' is better than `remove: true`

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useOnDeleteDatetime.ts]: Yes, `action` is more clear

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useUpdateDatetimeCache.ts [hunk 4]
```diff
@@ -1,15 +1,33 @@
...
 const useUpdateDatetimeCache = (): CacheUpdaterFn => {
 	const queryOptions = useDatetimeQueryOptions();
 
-	const updateDatetimeCache = ({ proxy, datetimes, datetime, remove = false }: CacheUpdaterFnArgs): void => {
+	const updateDatetimeCache = ({ proxy, datetimes, datetime, action }: CacheUpdaterFnArgs): void => {
 		const { nodes = [] } = datetimes;
-		// remove from or add to the list
-		const newNodes = remove ? nodes.filter(({ id }: Datetime) => id !== datetime.id) : [...nodes, datetime];
+		let newNodes: Array<Datetime>, datetimeIndex: number;
+		switch (action) {
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useUpdateDatetimeCache.ts]: no `default` today :crying_cat_face:

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useUpdateDatetimeCache.ts [hunk 5]
```diff
@@ -1,15 +1,33 @@
...
+			case 'add':
+				newNodes = [...nodes, datetime];
+				break;
+			case 'update':
+				// find the index of the datetime to update
+				datetimeIndex = findIndex(entityHasGuid(datetime.id), nodes);
+				// if datetime exists
+				if (datetimeIndex >= 0) {
+					newNodes = update(datetimeIndex, datetime, nodes);
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useUpdateDatetimeCache.ts]: love this

Comment (nicolad) [reply assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useUpdateDatetimeCache.ts]: also there is https://ramdajs.com/docs/#when and related for conditionals

File: assets/src/domain/eventEditor/services/apollo/mutations/prices/useUpdatePriceCache.ts [hunk 6]
```diff
@@ -1,15 +1,36 @@
...
+				if (priceIndex >= 0) {
+					newNodes = update(priceIndex, price, nodes);
+				}
+				break;
+			case 'remove':
+				newNodes = nodes.filter(({ id }) => id !== price.id);
+				break;
+			default:
+				newNodes = nodes;
+				break;
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/prices/useUpdatePriceCache.ts]: :+1:

File: assets/src/domain/eventEditor/services/apollo/queries/index.ts [hunk 7]
```diff
@@ -1,30 +1,12 @@
-import { GET_DATETIME, GET_DATETIMES } from './datetimes';
-import { GET_TICKET, GET_TICKETS } from './tickets';
-import { GET_PRICE, GET_PRICES } from './prices';
-import { GET_PRICE_TYPE, GET_PRICE_TYPES } from './priceTypes';
-import { GET_CURRENT_USER, GET_GENERAL_SETTINGS } from '../../../../shared/services/apollo/queries';
+// import { GET_DATETIME, GET_DATETIMES } from './datetimes';
+// import { GET_TICKET, GET_TICKETS } from './tickets';
+// import { GET_PRICE, GET_PRICES } from './prices';
+// import { GET_PRICE_TYPE, GET_PRICE_TYPES } from './priceTypes';
+// import { GET_CURRENT_USER, GET_GENERAL_SETTINGS } from '../../../../shared/services/apollo/queries';
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/queries/index.ts]: these will be used later?

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/services/apollo/queries/index.ts]: Oh, that needs to go

**Pull request merged**